packages/database/src/changeHandlers/EntityHierarchyCacher.js

```diff
@@ -52,8 +52,10 @@ export class EntityHierarchyCacher {
     old_record: oldRecord,
     new_record: newRecord,
   }) => {
-    if (oldRecord.canonical_types === newRecord.canonical_types) {
-      return null; // if the canonical types are the same, the change won't invalidate the cache
+    if (oldRecord && newRecord) {
```

Comment:
This is @edmofro code, not sure if okay to touch, someone should definitely have a look:

was getting this bug:
```error: TypeError: Cannot read property 'canonical_types' of null```

Reply:
```suggestion
    if (oldRecord?.canonical_types === newRecord?.canonical_types) {
```

Reply:
Good catch team, that's a bug from me! @nick-beyondessential nice syntax, recommend you use that @PShoe :-)

Reply:
Actually there is a slight logic difference in the syntax, and while the difference in behaviour is extremely edge casey, it makes me lean towards your original syntax for safety.

Logic difference explanation:
While the default for the column `canonical_types` is `{}`, there isn't an explicit `NOT NULL` constraint. That means you could get a situation where we're e.g. deleting a hierarchy, and for some reason its `canonical_types` had been set to `NULL`, in which which case you'd get:

`newRecord`: `null` because it is a delete, so `newRecord?.canonical_types` evaluates to `null`
`oldRecord`: `{ ..., canonical_types: null }`, so `oldRecord?.canonical_types` also evaluates to `null`

In which case the condition thinks that there is no change, and we don't invalidate the cache, which is incorrect. 

In fact, in this case we'd also be deleting the relation records etc. so the invalidation should happen as a byproduct of other actions, but just to avoid ever having to think about this again, @PShoe let's go with the way you have it!

Reply:
Thanks all! @nick-beyondessential / @edmofro